localstack-core/localstack/services/dynamodbstreams/v2/provider.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from localstack.aws import handlers
+from localstack.aws.api import RequestContext, ServiceRequest, ServiceResponse, handler
+from localstack.aws.api.dynamodbstreams import (
+    DescribeStreamInput,
+    DescribeStreamOutput,
+    DynamodbstreamsApi,
+    GetRecordsInput,
+    GetRecordsOutput,
+    GetShardIteratorInput,
+    GetShardIteratorOutput,
+    ListStreamsInput,
+    ListStreamsOutput,
+)
+from localstack.services.dynamodb.server import DynamodbServer
+from localstack.services.dynamodb.utils import modify_ddblocal_arns
+from localstack.services.dynamodb.v2.provider import DynamoDBProvider
+from localstack.services.plugins import ServiceLifecycleHook
+from localstack.utils.aws.arns import parse_arn
+
+LOG = logging.getLogger(__name__)
+
+
+class DynamoDBStreamsProvider(DynamodbstreamsApi, ServiceLifecycleHook):
+    def __init__(self):
+        self.server = DynamodbServer.get()
+
+    def on_after_init(self):
+        # add response processor specific to ddblocal
+        handlers.modify_service_response.append(self.service, modify_ddblocal_arns)
+
+    def forward_request(
+        self, context: RequestContext, service_request: ServiceRequest = None
+    ) -> ServiceResponse:
+        """
+        Forward a request to DynamoDB Local.
+        """
+        DynamoDBProvider.prepare_request_headers(
+            context.request.headers, account_id=context.account_id, region_name=context.region
+        )
+        return self.server.proxy(context, service_request)
+
+    def modify_stream_arn_for_ddb_local(self, stream_arn: str) -> str:
+        parsed_arn = parse_arn(stream_arn)
+
+        return f"arn:aws:dynamodb:ddblocal:000000000000:{parsed_arn['resource']}"
```

Comment:
Any idea how well this plays when non-default account/region is used in the request context?

Would the ARN in the response returned by DDBLocal need to be updated too?

Reply:
No idea no, I'll give it a try quickly to see, thanks for pointing it out!

Reply:
The ARN returned by DDBLocal are being updated by the following service response handler: `localstack.services.dynamodb.utils.modify_ddblocal_arns`.

I've checked and the needed StreamArn is always the `000000000000` account id and `ddblocal` region, works properly with non default account and region 👍 thanks for checking!